### PR TITLE
Implementation of Folder Jumping

### PR DESCRIPTION
**Pull request by sriio**
Simple implementation of [ Folder Jumping](https://github.com/blog/1877-folder-jumping)

Comes with this [PR](https://github.com/go-gitea/gitea/pull/511)

### CODE REVIEW

**Comment by andreynering** [comment]
LGTM

File: tree_entry.go [hunk 1]
```diff
@@ -94,6 +94,25 @@ func (te *TreeEntry) Blob() *Blob {
...
+// GetSubJumpablePathName return the full path of subdirectory jumpable ( contains only one directory )
+func (te *TreeEntry) GetSubJumpablePathName() string {
+	if te.IsSubModule() || !te.IsDir() {
+		return ""
+	}
+	tree, err := te.ptree.SubTree(te.name)
+	if err != nil {
+		return te.name
+	}
+	entries, err := tree.ListEntries()
```
Comment (Bwko) [inline tree_entry.go]: What happens when err != nil?

Comment (sriio) [inline tree_entry.go]: If err != nil, entiers variable will be nil. 
So i managed that simply with checking lenght of entries ( if len(entries) == 1 ).

Comment (Bwko) [inline tree_entry.go]: Then it's better to _ (underscore) the error

Comment (sriio) [inline tree_entry.go]: You right. I fix it.

**Comment by Bwko** [comment]
Some nitpicking but could you squash your 2 commits into 1?

**Comment by bkcsoft** [comment]
LGTM I'll squash

**Pull request merged**